### DISCUSSION TITLE
Roadmap for 25.01

### DISCUSSION THREAD

**Discussion by fballiano** [MahoCommerce/maho#52] [Roadmaps]
Task                                                                                                                                                  | Status
------------------------------------------------------------------------------------------ | ------
[New dashboard charts](https://x.com/fballiano/status/1865012399444521440)                   | ✅
Prototypejs-free frontend                                                                                                               | ✅
[RWD: New design for star rating](https://github.com/MahoCommerce/maho/pull/76)           | ✅
[2FA for backend](https://github.com/MahoCommerce/maho/pull/83)                                      | ✅
[Remove ExtJS](https://github.com/MahoCommerce/maho/pull/81)                                          | ✅
[EAV attribute editor](https://github.com/MahoCommerce/maho/pull/25)                                | in progress
Advanced grids for backend                                                                                                           | in progress
Moving "awesome list" of modules to maho's website                                                                | 

**Want more?**  
Consider [getting involved](https://mahocommerce.com/community/get-involved) or [sponsoring](https://mahocommerce.com/sponsors) ❤️

**Comment by alessandro-magi** [comment]
Hi Fabrizio,
I'm following your project, here are some ideas that could be useful to fill the gap with some solutions like Shopify.

- Drag & drop content editor, working on product cards, category and cms pages
- Drag & drop editor for transactional communications
- Order tracking page to be improved, like in Shopify
- In Shopify, if a guest user decides to register, he can recover the list of orders made, even if he has not previously registered and/or logged in
- In Magento, if a user enters the wrong email address, he does not have the possibility to change the email associated with the order (especially if he is a guest user) and have the order confirmation sent again by the system. Therefore, subsequent communications are not sent either.
- In the order comments section, it would be useful to be able to choose the order status, especially if one uses custom statuses
- In the order comments section, when you want to send a communication to the customer and it must remain in the backend, it would be useful to have a classic editor, such as tinymce, so as to format the text, instead of sending unformatted messages.
- Accept web, gif, mpeg video as formats for the gallery, obviously the gallery must be updated and support this type of multimedia files.

Thanks
Alessandro

**Reply by fballiano** [reply]
Hi @alessandro-magi! Thanks for these great ideas!

> * Drag & drop content editor, working on product cards, category and cms pages
> * Drag & drop editor for transactional communications

this is on our radar but it's a super big work, we're evaluating a few open source project to do that, let's see what happens

> * Order tracking page to be improved, like in Shopify

do you have a screenshot or a description of this?

> * In Shopify, if a guest user decides to register, he can recover the list of orders made, even if he has not previously registered and/or logged in
> * In Magento, if a user enters the wrong email address, he does not have the possibility to change the email associated with the order (especially if he is a guest user) and have the order confirmation sent again by the system. Therefore, subsequent communications are not sent either.
> * In the order comments section, when you want to send a communication to the customer and it must remain in the backend, it would be useful to have a classic editor, such as tinymce, so as to format the text, instead of sending unformatted messages.

these shouldn't be too hard, I'll keep it in mind

> * In the order comments section, it would be useful to be able to choose the order status, especially if one uses custom statuses

status is already changeable while commenting an order (from backend), state instead is not because that's driven by other logics

> * Accept web, gif, mpeg video as formats for the gallery, obviously the gallery must be updated and support this type of multimedia files.

webp/gif should already be supported (do you mean animated gifs?), I'll check about videos as soon as possible

**Reply by alessandro-magi** [reply]
Thank you @fballiano

> > * Order tracking page to be improved, like in Shopify
> 
> do you have a screenshot or a description of this?
> 
![Screenshot 2024-11-06 alle 12 15 35](https://github.com/user-attachments/assets/315e9cbf-8c33-44f0-8833-dbdff9ecbf8c)

This is the Frontend screen of one of my Shopify sites, when the tracking is inserted via backend or via API, by inserting the courier who must manage the shipment, for the end user in addition to sending the transactional email, with the tracking and the clickable link to track the shipment, on the frontend side, in the customer account, there is this part related to the status of the shipment.
